## Explore Data in the Global Emissions Dataset

### Import

In [3]:
import pandas as pd
import numpy as np

### Load Data

In [10]:
dataFrame = pd.read_csv('Data/GCB2022v27_MtCO2_flat.csv')
print(dataFrame.describe())
print(dataFrame.info())

               Year         Total          Coal           Oil           Gas  \
count  63104.000000  62904.000000  21744.000000  21717.000000  21618.000000   
mean    1885.500000     55.224788     73.968916     55.760624     23.504285   
std       78.519728    824.845435    598.986992    519.034563    247.674772   
min     1750.000000      0.000000      0.000000      0.000000      0.000000   
25%     1817.750000      0.000000      0.000000      0.091600      0.000000   
50%     1885.500000      0.000000      0.271852      1.044240      0.000000   
75%     1953.250000      0.549342      6.736411      8.339752      0.581628   
max     2021.000000  37123.850352  15051.512770  12345.653374   7921.829472   

             Cement       Flaring        Other    Per Capita  
count  20814.000000  21550.000000  1620.000000  18974.000000  
mean       4.330443      1.712695    10.951389      4.413363  
std       50.305770     16.727067    39.034073     17.432815  
min        0.000000      0.000000   

### Notes on data

Mean year of 1885 is the first interesting that pops out. This obviously relates to the industrial adoption of coal production, I am interested to see what countries differ from this average and possibly why.

I am also intrested to see when and by how much the different countries phased out coal for oil, or if they have at all. I also wonder if any of this data can be correlated to ideology. 

### Exploring the Year

In [17]:
dataFrame.groupby(['Country']).agg(min)

ISO 3166-1 alpha-3  Year  Total      Coal       Oil  \
Country                                                                         
Afghanistan                              AFG  1750    0.0  0.003664  0.000000   
Albania                                  ALB  1750    0.0  0.000000  0.007328   
Algeria                                  DZA  1750    0.0  0.000000  0.000000   
Andorra                                  AND  1750    0.0  0.000000  0.406704   
Angola                                   AGO  1750    0.0  0.000000  0.127068   
...                                      ...   ...    ...       ...       ...   
Viet Nam                                 VNM  1750    0.0  0.179536  0.000000   
Wallis and Futuna Islands                WLF  1750    0.0  0.000000  0.018320   
Yemen                                    YEM  1750    0.0  0.000000  0.040304   
Zambia                                   ZMB  1750    0.0  0.131904  0.095425   
Zimbabwe                                 ZWE  1750    0.0  0.113584  0.000000   

                           Gas   Cement  Flaring  Other  Per Capita  
Country                                                              
Afghanistan                0.0  0.00000      0.0    NaN    0.011266  
Albania                    0.0  0.00000      0.0    NaN    0.236891  
Algeria                    0.0  0.00000      0.0    NaN    0.408457  
Andorra                    0.0  0.00000      0.0    NaN    0.000000  
Angola                     0.0  0.00000      0.0    NaN    0.041728  
...                        ...      ...      ...    ...         ...  
Viet Nam                   0.0  0.00000      0.0    NaN    0.050149  
Wallis and Futuna Islands  0.0  0.00000      0.0    NaN    0.000000  
Yemen                      0.0  0.00000      0.0    NaN    0.008427  
Zambia                     0.0  0.02405      0.0    NaN    0.180375  
Zimbabwe                   0.0  0.00000      0.0    NaN    0.615130  

[232 rows x 10 columns]